pkg/services/vmoperator/vmopmachine.go

```diff
@@ -38,12 +38,14 @@ import (
 	vmwarev1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/vmware/v1beta1"
 	capvcontext "sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context/vmware"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services"
 	infrautilv1 "sigs.k8s.io/cluster-api-provider-vsphere/pkg/util"
 )
 
 // VmopMachineService reconciles VM Operator VM.
 type VmopMachineService struct {
-	Client client.Client
+	Client          client.Client
+	NetworkProvider services.NetworkProvider
```

Comment:
Wondering if we should have a bool here instead of a full blown network provider.

Reply:
That's a nice idea.